core/execution/market.go

```diff
@@ -2456,6 +2456,9 @@ func (m *Market) AmendOrderWithIDGenerator(
 
 	if !m.as.InAuction() {
 		m.checkForReferenceMoves(ctx, allUpdatedOrders, false)
+	} else {
+		// make sure that the amended orders are updated in the liquidity engine too
+		m.liquidity.SynchronisePartyOrders(party, allUpdatedOrders)
```

Comment:
Isn’t the Update method on liquidity sufficient for that? Can’t remember what parameters it takes tho.

Reply:
Actually yea I think we should ideally use the update method or so, which will also recreate order if needed?

Reply:
Its the same parameter but it calls stuff like `e.createOrUpdateForParty()` which I wasn't sure if we wanted to do when we literally just want to make the orders the same. 

Theres was also `updatePartyOrders()` which I tried but there is some check in there which we didn't satisfy and then didn't update.

Reply:
Alright, I’ll think a little bit over it if you don’t mind before we merge.

Reply:
Yea sure thing, rather it be done properly than something that just works.

Reply:
@wwestgarth @jeremyletang I wonder if we're not creating a problem to ourselves, i.e. in trying to keep two instances in sync instead of just having one source of truth being the order book and not keeping another copy elsewhere but querying the order book for it when needed.

Reply:
FYI, https://github.com/vegaprotocol/vega/issues/6612 is the same issue. Snapshot 82000 which is the one we try to restore from has the order in inconsistent state between the order book and the liquidity party orders. When the node is loading this snapshot, the inconsistency is corrected via the reconcile() but now it's inconsistent with all other nodes. 

on the left is the liquidityPartiesOrders party orders from snapshot 82000. On the right is the same from a snapshot taken right after the loading to memory of 82000.  
<img width="1489" alt="snapshot_lp_party_orders_vs_after_loading" src="https://user-images.githubusercontent.com/6655556/198287233-1a00197e-4782-4c34-85fd-2c6d98e11b09.png">

on the left is the order book order in snapshot 82000 and on the right, the same after loading. 
<img width="1540" alt="snapshot_orderbook_vs_after_loading" src="https://user-images.githubusercontent.com/6655556/198287227-35ad343d-f2db-413b-a728-bb1084913c3a.png">

You can see that in the snapshot the order in party orders is in incorrect state, which is corrected when loaded in the snapshot. When running with this branch it leads to the expected panic.

Reply:
Agree that we should just be asking the orderbook (unless there was a good reason why it wasn't originally done this way?). At a glance it doesn't look to be a hard change to make.